flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedApproximateSubpartitionView.java

```diff
@@ -0,0 +1,27 @@
+package org.apache.flink.runtime.io.network.partition;
+
+import static org.apache.flink.util.Preconditions.checkState;
+
+public class PipelinedApproximateSubpartitionView extends PipelinedSubpartitionView {
+
+	PipelinedApproximateSubpartitionView(PipelinedApproximateSubpartition parent, BufferAvailabilityListener listener) {
+		super(parent, listener);
+	}
+
+	@Override
+	public void releaseAllResources() {
+		if (isReleased.compareAndSet(false, true)) {
+			// The view doesn't hold any resources and the parent cannot be restarted. Therefore,
+			// it's OK to notify about consumption as well.
+			checkState(parent instanceof PipelinedApproximateSubpartition);
+			((PipelinedApproximateSubpartition) parent).releaseView();
```

Comment:
It might also be an option to replicate the field in this class with `PipelinedApproximateSubpartition` (a slightly different name would also be good like `approximateParent`).

Reply:
Yeah, I was also thinking of that. The problem is that I have to change all the common methods in the superclass that uses `parent` to `approximateParent` => a lot of redundant code.

`releaseView()` is one of the few classes that are different from the superclass.